src/System.Net.HttpListener/src/System/Net/HttpListener.cs

```diff
@@ -112,6 +112,158 @@ public HttpListenerPrefixCollection Prefixes
             }
         }
 
+        internal void AddPrefix(string uriPrefix)
+        {
+            if (NetEventSource.IsEnabled) NetEventSource.Enter(this, $"uriPrefix:{uriPrefix}");
+            string registeredPrefix = null;
+            try
+            {
+                if (uriPrefix == null)
+                {
+                    throw new ArgumentNullException(nameof(uriPrefix));
+                }
+                CheckDisposed();
+                if (NetEventSource.IsEnabled) NetEventSource.Info(this, "uriPrefix:" + uriPrefix);
+                int i;
+                if (string.Compare(uriPrefix, 0, "http://", 0, 7, StringComparison.OrdinalIgnoreCase) == 0)
+                {
+                    i = 7;
+                }
+                else if (string.Compare(uriPrefix, 0, "https://", 0, 8, StringComparison.OrdinalIgnoreCase) == 0)
+                {
+                    i = 8;
+                }
+                else
+                {
+                    throw new ArgumentException(SR.net_listener_scheme, nameof(uriPrefix));
+                }
+                bool inSquareBrakets = false;
+                int j = i;
+                while (j < uriPrefix.Length && uriPrefix[j] != '/' && (uriPrefix[j] != ':' || inSquareBrakets))
+                {
+                    if (uriPrefix[j] == '[')
+                    {
+                        if (inSquareBrakets)
+                        {
+                            j = i;
+                            break;
+                        }
+                        inSquareBrakets = true;
+                    }
+                    if (inSquareBrakets && uriPrefix[j] == ']')
+                    {
+                        inSquareBrakets = false;
+                    }
+                    j++;
+                }
+                if (i == j)
+                {
+                    throw new ArgumentException(SR.net_listener_host, nameof(uriPrefix));
+                }
+                if (uriPrefix[uriPrefix.Length - 1] != '/')
+                {
+                    throw new ArgumentException(SR.net_listener_slash, nameof(uriPrefix));
+                }
+                registeredPrefix = uriPrefix[j] == ':' ? String.Copy(uriPrefix) : uriPrefix.Substring(0, j) + (i == 7 ? ":80" : ":443") + uriPrefix.Substring(j);
+                fixed (char* pChar = registeredPrefix)
+                {
+                    i = 0;
+                    while (pChar[i] != ':')
+                    {
+                        pChar[i] = (char)CaseInsensitiveAscii.AsciiToLower[(byte)pChar[i]];
+                        i++;
+                    }
+                }
+                if (NetEventSource.IsEnabled) NetEventSource.Info(this, "mapped uriPrefix:" + uriPrefix + " to registeredPrefix:" + registeredPrefix);
+                if (_state == State.Started)
+                {
+                    if (NetEventSource.IsEnabled) NetEventSource.Info(this, "Calling Interop.HttpApi.HttpAddUrl[ToUrlGroup]");
```

Comment:
This tracing should be inside the AddPrefixCore, as it's specific to the Windows implementation.